wrappers/kn/build.gradle.kts

```diff
@@ -0,0 +1,231 @@
+                implementation(libs.test.korlibs.korim)
+            }
+        }
+        if (hostOs == "Mac OS X") {
+            appleTest.get().dependsOn(nonAndroidNativeTest)
+        }
+        linuxTest.get().dependsOn(nonAndroidNativeTest)
+        mingwTest.get().dependsOn(nonAndroidNativeTest)
+    }
+}
+
+krossCompile {
+    libraries {
+        val zxingCpp by creating {
+            sourceDir = file("../../core").absolutePath
+            outputPath = ""
+            libraryArtifactNames = listOf("libZXing.a")
+
+            cinterop {
+                packageName = "zxingcpp.cinterop"
+                headers = listOf("$sourceDir/src/zxing-c.h")
+            }
+            cmake.apply {
+                val buildPath = project.layout.buildDirectory.dir("cmake").get().asFile.absolutePath +
+                        "/{projectName}/{targetName}"
+                configParams {
+                    buildDir = buildPath
+                }
+                configParams += (ModifiablePlatformEntriesImpl().apply {
+                    buildType = "Release"
+                    buildSharedLibs = false
+                } + CustomCMakeCacheEntries(
+                    mapOf(
+                        "BUILD_C_API" to "ON",
+                        "BUILD_WRITERS" to "OFF",
+                    )
+                )).asCMakeParams
+                buildParams {
+                    buildDir = buildPath
+                    config = "Release"
+                }
+                buildParams += CustomCMakeParams(listOf("-j16"))
+            }
+
+            androidNativeX64.konan()
+            androidNativeX86.konan()
+            androidNativeArm32.konan()
+            androidNativeArm64.konan()
+
+            linuxX64.konan {
+                cmake {
+                    configParams += CustomCMakeCacheEntries(
+                        mapOf(
+                            "CMAKE_CXX_STANDARD" to "17",
+                        )
+                    ).asCMakeParams
+                }
+            }
+            linuxArm64.konan {
+                cmake {
+                    configParams += CustomCMakeCacheEntries(
+                        mapOf(
+                            "CMAKE_CXX_STANDARD" to "17",
+                        )
+                    ).asCMakeParams
+                }
+            }
+            mingwX64.konan {
+                cinterop {
+                    linkerOpts += "-Wl,-Xlink=-force:multiple"
+                }
+            }
+
+            if (hostOs == "Mac OS X") {
+                iosX64.konan()
+                iosArm64.konan()
+                iosSimulatorArm64.konan()
+                macosX64.konan()
+                macosArm64.konan()
+                watchosX64.konan()
+                watchosArm32.konan()
+                watchosArm64.konan()
+                watchosSimulatorArm64.konan()
+                tvosX64.konan()
+                tvosArm64.konan()
+                tvosSimulatorArm64.konan()
```

Comment:
yes, the native binaries are shipped to mavencentral, but they are not stored in one package
They are store like following file structure.
```
[5.4M]  .m2/repository/io/github/zxing-cpp
├── [9.5K]  kotlin-native
│   ├── [5.2K]  2.2.1
│   │   └── [1.2K]  kotlin-native-2.2.1.pom
│   └── [ 339]  maven-metadata-local.xml
├── [ 60K]  zxing-cpp
│   ├── [ 56K]  2.2.1
│   │   ├── [2.5K]  zxing-cpp-2.2.1-kotlin-tooling-metadata.json
│   │   ├── [4.9K]  zxing-cpp-2.2.1-sources.jar
│   │   ├── [ 33K]  zxing-cpp-2.2.1.jar
│   │   ├── [ 10K]  zxing-cpp-2.2.1.module
│   │   └── [ 981]  zxing-cpp-2.2.1.pom
│   └── [ 335]  maven-metadata-local.xml
├── [735K]  zxing-cpp-androidnativearm32
│   ├── [730K]  2.2.1
│   │   ├── [676K]  zxing-cpp-androidnativearm32-2.2.1-cinterop-zxingCpp.klib
│   │   ├── [4.9K]  zxing-cpp-androidnativearm32-2.2.1-sources.jar
│   │   ├── [ 41K]  zxing-cpp-androidnativearm32-2.2.1.klib
│   │   ├── [2.9K]  zxing-cpp-androidnativearm32-2.2.1.module
│   │   └── [1.0K]  zxing-cpp-androidnativearm32-2.2.1.pom
│   └── [ 354]  maven-metadata-local.xml
├── [735K]  zxing-cpp-androidnativearm64
│   ├── [730K]  2.2.1
│   │   ├── [676K]  zxing-cpp-androidnativearm64-2.2.1-cinterop-zxingCpp.klib
│   │   ├── [4.9K]  zxing-cpp-androidnativearm64-2.2.1-sources.jar
│   │   ├── [ 41K]  zxing-cpp-androidnativearm64-2.2.1.klib
│   │   ├── [2.9K]  zxing-cpp-androidnativearm64-2.2.1.module
│   │   └── [1.0K]  zxing-cpp-androidnativearm64-2.2.1.pom
│   └── [ 354]  maven-metadata-local.xml
├── [734K]  zxing-cpp-androidnativex64
│   ├── [730K]  2.2.1
│   │   ├── [676K]  zxing-cpp-androidnativex64-2.2.1-cinterop-zxingCpp.klib
│   │   ├── [4.9K]  zxing-cpp-androidnativex64-2.2.1-sources.jar
│   │   ├── [ 41K]  zxing-cpp-androidnativex64-2.2.1.klib
│   │   ├── [2.8K]  zxing-cpp-androidnativex64-2.2.1.module
│   │   └── [1.0K]  zxing-cpp-androidnativex64-2.2.1.pom
│   └── [ 352]  maven-metadata-local.xml
├── [734K]  zxing-cpp-androidnativex86
│   ├── [730K]  2.2.1
│   │   ├── [676K]  zxing-cpp-androidnativex86-2.2.1-cinterop-zxingCpp.klib
│   │   ├── [4.9K]  zxing-cpp-androidnativex86-2.2.1-sources.jar
│   │   ├── [ 41K]  zxing-cpp-androidnativex86-2.2.1.klib
│   │   ├── [2.8K]  zxing-cpp-androidnativex86-2.2.1.module
│   │   └── [1.0K]  zxing-cpp-androidnativex86-2.2.1.pom
│   └── [ 352]  maven-metadata-local.xml
├── [835K]  zxing-cpp-linuxarm64
│   ├── [831K]  2.2.1
│   │   ├── [777K]  zxing-cpp-linuxarm64-2.2.1-cinterop-zxingCpp.klib
│   │   ├── [4.9K]  zxing-cpp-linuxarm64-2.2.1-sources.jar
│   │   ├── [ 41K]  zxing-cpp-linuxarm64-2.2.1.klib
│   │   ├── [2.8K]  zxing-cpp-linuxarm64-2.2.1.module
│   │   └── [1018]  zxing-cpp-linuxarm64-2.2.1.pom
│   └── [ 346]  maven-metadata-local.xml
├── [869K]  zxing-cpp-linuxx64
│   ├── [864K]  2.2.1
│   │   ├── [811K]  zxing-cpp-linuxx64-2.2.1-cinterop-zxingCpp.klib
│   │   ├── [4.9K]  zxing-cpp-linuxx64-2.2.1-sources.jar
│   │   ├── [ 41K]  zxing-cpp-linuxx64-2.2.1.klib
│   │   ├── [2.8K]  zxing-cpp-linuxx64-2.2.1.module
│   │   └── [1016]  zxing-cpp-linuxx64-2.2.1.pom
│   └── [ 344]  maven-metadata-local.xml
└── [819K]  zxing-cpp-mingwx64
    ├── [814K]  2.2.1
    │   ├── [761K]  zxing-cpp-mingwx64-2.2.1-cinterop-zxingCpp.klib
    │   ├── [4.9K]  zxing-cpp-mingwx64-2.2.1-sources.jar
    │   ├── [ 41K]  zxing-cpp-mingwx64-2.2.1.klib
    │   ├── [2.8K]  zxing-cpp-mingwx64-2.2.1.module
    │   └── [1016]  zxing-cpp-mingwx64-2.2.1.pom
    └── [ 344]  maven-metadata-local.xml

  16M used in 19 directories, 50 files
```

Reply:
opps! It seems there are problem naming artifacts, fix it later.

Reply:
Fixed in: https://github.com/zxing-cpp/zxing-cpp/pull/719/commits/e5db627c1c1b3146fd887faedd419b829d628a5a
Here is the new file structure:
```
[5.4M]  .m2/repository/io/github/zxing-cpp
├── [ 61K]  kotlin-native
│   ├── [ 57K]  2.2.1
│   │   ├── [2.5K]  kotlin-native-2.2.1-kotlin-tooling-metadata.json
│   │   ├── [4.8K]  kotlin-native-2.2.1-sources.jar
│   │   ├── [ 33K]  kotlin-native-2.2.1.jar
│   │   ├── [ 11K]  kotlin-native-2.2.1.module
│   │   └── [1.7K]  kotlin-native-2.2.1.pom
│   └── [ 339]  maven-metadata-local.xml
├── [734K]  kotlin-native-androidnativearm32
│   ├── [729K]  2.2.1
│   │   ├── [676K]  kotlin-native-androidnativearm32-2.2.1-cinterop-zxingCpp.klib
│   │   ├── [4.8K]  kotlin-native-androidnativearm32-2.2.1-sources.jar
│   │   ├── [ 40K]  kotlin-native-androidnativearm32-2.2.1.klib
│   │   ├── [2.9K]  kotlin-native-androidnativearm32-2.2.1.module
│   │   └── [1.7K]  kotlin-native-androidnativearm32-2.2.1.pom
│   └── [ 358]  maven-metadata-local.xml
├── [734K]  kotlin-native-androidnativearm64
│   ├── [729K]  2.2.1
│   │   ├── [676K]  kotlin-native-androidnativearm64-2.2.1-cinterop-zxingCpp.klib
│   │   ├── [4.8K]  kotlin-native-androidnativearm64-2.2.1-sources.jar
│   │   ├── [ 40K]  kotlin-native-androidnativearm64-2.2.1.klib
│   │   ├── [2.9K]  kotlin-native-androidnativearm64-2.2.1.module
│   │   └── [1.7K]  kotlin-native-androidnativearm64-2.2.1.pom
│   └── [ 358]  maven-metadata-local.xml
├── [734K]  kotlin-native-androidnativex64
│   ├── [729K]  2.2.1
│   │   ├── [676K]  kotlin-native-androidnativex64-2.2.1-cinterop-zxingCpp.klib
│   │   ├── [4.8K]  kotlin-native-androidnativex64-2.2.1-sources.jar
│   │   ├── [ 40K]  kotlin-native-androidnativex64-2.2.1.klib
│   │   ├── [2.9K]  kotlin-native-androidnativex64-2.2.1.module
│   │   └── [1.7K]  kotlin-native-androidnativex64-2.2.1.pom
│   └── [ 356]  maven-metadata-local.xml
├── [733K]  kotlin-native-androidnativex86
│   ├── [729K]  2.2.1
│   │   ├── [676K]  kotlin-native-androidnativex86-2.2.1-cinterop-zxingCpp.klib
│   │   ├── [4.8K]  kotlin-native-androidnativex86-2.2.1-sources.jar
│   │   ├── [ 40K]  kotlin-native-androidnativex86-2.2.1.klib
│   │   ├── [2.9K]  kotlin-native-androidnativex86-2.2.1.module
│   │   └── [1.7K]  kotlin-native-androidnativex86-2.2.1.pom
│   └── [ 356]  maven-metadata-local.xml
├── [834K]  kotlin-native-linuxarm64
│   ├── [830K]  2.2.1
│   │   ├── [777K]  kotlin-native-linuxarm64-2.2.1-cinterop-zxingCpp.klib
│   │   ├── [4.8K]  kotlin-native-linuxarm64-2.2.1-sources.jar
│   │   ├── [ 40K]  kotlin-native-linuxarm64-2.2.1.klib
│   │   ├── [2.8K]  kotlin-native-linuxarm64-2.2.1.module
│   │   └── [1.7K]  kotlin-native-linuxarm64-2.2.1.pom
│   └── [ 350]  maven-metadata-local.xml
├── [868K]  kotlin-native-linuxx64
│   ├── [864K]  2.2.1
│   │   ├── [811K]  kotlin-native-linuxx64-2.2.1-cinterop-zxingCpp.klib
│   │   ├── [4.8K]  kotlin-native-linuxx64-2.2.1-sources.jar
│   │   ├── [ 40K]  kotlin-native-linuxx64-2.2.1.klib
│   │   ├── [2.8K]  kotlin-native-linuxx64-2.2.1.module
│   │   └── [1.7K]  kotlin-native-linuxx64-2.2.1.pom
│   └── [ 348]  maven-metadata-local.xml
└── [818K]  kotlin-native-mingwx64
    ├── [814K]  2.2.1
    │   ├── [761K]  kotlin-native-mingwx64-2.2.1-cinterop-zxingCpp.klib
    │   ├── [4.8K]  kotlin-native-mingwx64-2.2.1-sources.jar
    │   ├── [ 40K]  kotlin-native-mingwx64-2.2.1.klib
    │   ├── [2.8K]  kotlin-native-mingwx64-2.2.1.module
    │   └── [1.7K]  kotlin-native-mingwx64-2.2.1.pom
    └── [ 348]  maven-metadata-local.xml

  16M used in 17 directories, 48 files
```

Reply:
> Very interesting. This naming conflict must mean that with the existing "android" package there is something wrong in the sens that the term "zxing-cpp" slipped though (instead of "android") and I should make the same change there that you just did here?
I do this because the publication is automatically configured by kotlin multiplatform plugin with prefix of project name, I don't know whether the android library plugin had done same. Maybe you can give a look to the file structure created when publishing android library to maven local?

> Also: where does the 'C' come from in `...cinterop-zxingCpp.klib`?
As far as I know, the c static library(and other metadata about c interoperation) is packaged in this file

Reply:
I thought the conflict that you were talking about must have been between this new kn data and the existing android one already in maven central but [there](https://repo1.maven.org/maven2/io/github/zxing-cpp/android/2.2.0/) is no mentioning of the string "zxing-cpp". Then I don't understand what you issue you have been fixing here but still believe your new file-structure is correct.

I was not wondering what is inside this .klib but rather where the capital 'C' inside `zxingCpp` comes from. If that is because you configured it that way somewhere, then please use the string `zxingcpp` or `zxing-cpp` for consistency.

Reply:
> I thought the conflict that you were talking about must have been between this new kn data and the existing android one already in maven central but [there](https://repo1.maven.org/maven2/io/github/zxing-cpp/android/2.2.0/) is no mentioning of the string "zxing-cpp". Then I don't understand what you issue you have been fixing here but still believe your new file-structure is correct.
At first I simply copied codes about publishing packages from android wrapper, and it resulted in an empty package `kotlin-native` published with real packages naming with project name `zxing-cpp`. This is not expected. (In the commit) I fix it by correct their names one by one.

> I was not wondering what is inside this .klib but rather where the capital 'C' inside `zxingCpp` comes from. If that is because you configured it that way somewhere, then please use the string `zxingcpp` or `zxing-cpp` for consistency.
I think the capital 'C' is caused by kotlin gradle plugin's internal stratagy converting '-' seprated name into lower camel case.

Reply:
Thanks for clarifying that.